### PR TITLE
Fix: add Target fail by fetch terraform-provider error

### LINKED ISSUE #3987: add Target fail by fetch terraform-provider error
**Describe the bug**

image

**To Reproduce**
1. go to VelaUX
2. go to menu Targets
3. click button Add Target
4. select one Project, such as Default
5. one error is raised

**Expected behavior**
1. no error is raised

**Screenshots**

**KubeVela Version**

CLI Version: v1.4.0-beta.1
Core Version: 1.4.0-beta.1
GitRevision: git-1d82378
GolangVersion: go1.17.10
**Cluster information**

**Additional context**

### PR DESCRIPTION
**Pull request by StevenLeiZhang**
Signed-off-by: StevenLeiZhang <zhangleiic@163.com>

### Description of your changes

Fixes #3987 

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/kubevela/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/kubevela/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

File: pkg/apiserver/domain/service/project.go [hunk 1]
```diff
@@ -495,7 +495,7 @@ func (p *projectServiceImpl) GetConfigs(ctx context.Context, projectName, config
 		// legacy providers
 		var providers = &terraformapi.ProviderList{}
 		if err := p.K8sClient.List(ctx, providers, client.InNamespace(types.DefaultAppNamespace)); err != nil {
-			return nil, err
+			log.Logger.Infof("fail to list legacy providers, err: %v", err)
```
[the remark this answers is not in the document]
Comment (StevenLeiZhang) [reply pkg/apiserver/domain/service/project.go:498]: I think this code block is used for legacy providers compatibility, so just log some message and continue to new code logic.

[the remark this answers is not in the document]
Comment (barnettZQG) [reply pkg/apiserver/domain/service/project.go:498]: No, this is core logic but depends on the terraform addon. we should ignore the no matches kind error before the terraform addon is installed.

[the remark this answers is not in the document]
Comment (StevenLeiZhang) [reply pkg/apiserver/domain/service/project.go]: > No, this is core logic but depends on the terraform addon. we should ignore the no matches kind error before the terraform addon is installed.

Ok, Updated.

**Review by wonderflow — APPROVED** [review]
LGTM, thanks

File: pkg/apiserver/domain/service/project.go [hunk 2]
```diff
@@ -495,7 +495,11 @@ func (p *projectServiceImpl) GetConfigs(ctx context.Context, projectName, config
 		// legacy providers
 		var providers = &terraformapi.ProviderList{}
 		if err := p.K8sClient.List(ctx, providers, client.InNamespace(types.DefaultAppNamespace)); err != nil {
-			return nil, err
+			// this logic depends on the terraform addon, ignore the no matches kind error before the terraform addon is installed.
+			if !strings.Contains(err.Error(), "no matches for kind \"Provider\" in version \"terraform.core.oam.dev") {
```
Comment (barnettZQG) [inline pkg/apiserver/domain/service/project.go:499]: `meta.IsNoMatchError(err)` Replace with this code is more clear?

Comment (StevenLeiZhang) [reply pkg/apiserver/domain/service/project.go]: > `meta.IsNoMatchError(err)` Replace with this code is more clear?

Thanks. Updated.

**Review by barnettZQG — APPROVED** [review]
@StevenLeiZhang These changes need manual backport for the 1.3-release branch.

**Pull request merged**